File: 495. Teemo Attacking/495. Teemo Attacking.py
class Solution(object):
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """
        if timeSeries == []:
            return 0
        cnt = duration #the last one
        for i in xrange(len(timeSeries) - 1):
            if timeSeries[i] + duration < timeSeries[i + 1]:
                cnt += duration
            else:
                cnt += (timeSeries[i + 1] - timeSeries[i])
        return cnt

File: 495. Teemo Attacking/495. Teemo Attacking.cs
public class Solution {
    public int FindPoisonedDuration(int[] timeSeries, int duration) {
        int cnt = duration, l = timeSeries.Length - 1;
        if (l == -1) {
            return 0;
        }
        for (int i = 0; i < l; i++) {
            if (timeSeries[i] + duration < timeSeries[i + 1]) {
                cnt += duration;
            } else {
                cnt += (timeSeries[i + 1] - timeSeries[i]);
            }
        }
        return cnt;
    }
}
